src/pages/version/TaskDuration.tsx

```diff
@@ -0,0 +1,150 @@
+import { useEffect } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { useParams, useLocation } from "react-router-dom";
+import { useVersionAnalytics } from "analytics";
+import { PageSizeSelector } from "components/PageSizeSelector";
+import { Pagination } from "components/Pagination";
+import { ResultCountLabel } from "components/ResultCountLabel";
+import { TableControlOuterRow, TableControlInnerRow } from "components/styles";
+import { pollInterval } from "constants/index";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import { PatchTasksQuery, PatchTasksQueryVariables } from "gql/generated/types";
+import { GET_PATCH_TASKS } from "gql/queries";
+import { usePolling } from "hooks";
+import { useUpdateURLQueryParams } from "hooks/useUpdateURLQueryParams";
+import { PatchTasksQueryParams } from "types/task";
+import { queryString, url } from "utils";
+import { TaskDurationTable } from "./taskDuration/TaskDurationTable";
+
+const { parseQueryString, parseSortString, getString } = queryString;
+
+const { getPageFromSearch, getLimitFromSearch } = url;
+interface Props {
+  taskCount: number;
+}
+
+export const TaskDuration: React.VFC<Props> = ({ taskCount }) => {
+  const { id: versionId } = useParams<{ id: string }>();
+
+  const { search } = useLocation();
+  const { sendEvent } = useVersionAnalytics(versionId);
+  const dispatchToast = useToastContext();
+
+  const updateQueryParams = useUpdateURLQueryParams();
+  const queryVariables = getQueryVariables(search, versionId);
+  const noQueryVariables = !search.length;
+
+  const { limit, page } = queryVariables;
+  const defaultSortMethod = "STATUS:ASC;BASE_STATUS:DESC;DURATION:DESC";
+
+  const { sorts: allSorts } = parseQueryString(search);
+  const sortOrders = parseSortString(allSorts);
+
+  useEffect(() => {
+    if (noQueryVariables) {
+      updateQueryParams({
+        sorts: defaultSortMethod,
+      });
+    }
+  }, [search]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const { data, startPolling, stopPolling } = useQuery<
+    PatchTasksQuery,
+    PatchTasksQueryVariables
+  >(GET_PATCH_TASKS, {
+    variables: queryVariables,
+    skip: noQueryVariables,
+    pollInterval,
+    fetchPolicy: "network-only",
+    nextFetchPolicy: "cache-and-network",
```

Comment:
It is because when I used `cache-first`, the tasks weren't being updated after reconfiguring a patch

https://user-images.githubusercontent.com/47064971/163219854-fa932cd7-c2fd-4b32-9b87-1905ccc54d3a.mov

Reply:
I'm not completely sure even after reading the docs but I think nextFetchPolicy will actually read updates from cache and make a network request for the next poll interval. If this is the case, we may be okay omitting the cache read if another component isn't updating the cache while TaskDuration is being rendered.